bitcoin/src/psbt/mod.rs

```diff
@@ -44,15 +44,55 @@ use self::map::Map;
 /// Partially signed transaction, commonly referred to as a PSBT.
 pub type Psbt = PartiallySignedTransaction;
 
+/// Future version of PSBT this library can't parse
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct FutureVersionError(u32);
+
+impl fmt::Display for FutureVersionError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str("Unknown future PSBT version ")?;
+        fmt::Display::fmt(&self.0, f)
```

Comment:
Sadly, this doesn't actually do what you expect. If one selects padding it will result into something like `Unknown future PSBT version                              42` - the spaces will be added in front of the number.

If you don't want to bother with it I suggest just using `write!()` to disable formatting options. We could add them later.

Reply:
What do you think I expect vs what this does? I haven't been able to get this to do something unexpected. What do you mean?

What does padding with spaces do? It does what I'd expect so far 🤔

```rust
let e = FutureVersionError(123);
println!("{number:0>width$}", number=123, width=6);
println!("{number:0>width$}", number=e, width=6);
println!("{number: >width$}", number=e, width=6);
```

produces

```
000123
Unknown future PSBT version 000123
Unknown future PSBT version    123
```

Reply:
You should *not* expect that output. :) Width is intended to work with the whole output, not just part of it. So it should produce `Unknown future PSBT version 123` in both cases. Then this:

```rust
println!("{number: <width$}", number=e, width=34);
```

should produce this:

```
   Unknown future PSBT version 123
```

And zero-padding should be disabled because it doesn't make sense (the message is not integer).